lib/Router.js

```diff
@@ -250,6 +250,20 @@ Router.prototype.handle = function(request, connection) {
         return request.error(404, "No handlers available for resource " + resourceId);
     }
 
+    if (!!~[Request.METHOD_PATCH, Request.METHOD_POST, Request.METHOD_PUT].indexOf(method)) {
+        var validate = jsen(_.extend({additionalProperties: false}, route.schema));
+        var resource = request.getResource();
+        if (resource) {
+            delete resource.$id;
```

Comment:
You may need to recursively scan the resource for these special properties. And it would be better to delete all properties that start with `$` in case we add more (`$etag` may be coming soon).